Clarify serving default documents with MapStaticAssets and document the static assets manifest (#37403)
dotnet/AspNetCore.Docs: aspnetcore/fundamentals/static-files.md

* [Set HTTP response headers](#set-http-response-headers)
* [Serving files from disk or embedded resources, or other locations](#serve-files-from-multiple-locations)
* [Directory browsing](#directory-browsing)

> * [Serve default documents](#serve-default-documents) (requires a call to <xref:Microsoft.AspNetCore.Builder.StaticFileExtensions.UseStaticFiles%2A> when using <xref:Microsoft.AspNetCore.Builder.StaticAssetsEndpointRouteBuilderExtensions.MapStaticAssets%2A>)

Reviewer: ```suggestion
* [Serve default documents](#serve-default-documents) (with <xref:Microsoft.AspNetCore.Builder.StaticAssetsEndpointRouteBuilderExtensions.MapStaticAssets%2A>, requires a call to <xref:Microsoft.AspNetCore.Builder.StaticFileExtensions.UseStaticFiles%2A>)
```
Was a bit awkward, shortened a bit.

app.MapStaticAssets();
```

:::moniker range=">= aspnetcore-9.0"

## Static assets manifest

> <xref:Microsoft.AspNetCore.Builder.StaticAssetsEndpointRouteBuilderExtensions.MapStaticAssets%2A> serves assets from a *static assets manifest* rather than by scanning the [web root](xref:fundamentals/index#web-root) at runtime. The manifest is generated at build and publish time and records the static web assets discovered for the app, along with metadata such as content fingerprints, `Content-Type` headers, caching headers, and the precomputed compressed representations (Gzip and Brotli). At runtime, `MapStaticAssets` reads the manifest, registers an endpoint for each asset, and serves the optimized responses.

Reviewer: ```suggestion
<xref:Microsoft.AspNetCore.Builder.StaticAssetsEndpointRouteBuilderExtensions.MapStaticAssets%2A> serves assets from a *static assets manifest* rather than by scanning the [web root](xref:fundamentals/index#web-root) at runtime. The manifest is generated at build and publish time and records the static web assets discovered for the app, along with metadata such as content fingerprints, `Content-Type` headers, caching headers, and the precomputed compressed representations ([Gzip](https://tools.ietf.org/html/rfc1952) and [Brotli](https://tools.ietf.org/html/rfc7932)). At runtime, `MapStaticAssets` reads the manifest, registers an endpoint for each asset, and serves the optimized responses.
```
Added links.

> The manifest is generated in the build output directory. By default, its file name is based on the app's <xref:Microsoft.Extensions.Hosting.IHostEnvironment.ApplicationName%2A> (for example, `{ASSEMBLY NAME}.staticwebassets.endpoints.json`, where the `{ASSEMBLY NAME}` placeholder is the app's assembly name). To provide a manifest from a different location, see the [Provide a custom static files manifest](#provide-a-custom-static-files-manifest) section.

Reviewer: This copilot suggestion would only half fix the problem.  The core inaccuracy is that the file name is derived from the MSBuild AssemblyName (a build-time artifact), while IHostEnvironment.ApplicationName is a configurable runtime value that only defaults to the assembly name. Copilot's suggestion just softens "based on ApplicationName" to "usually matches ApplicationName" but keeps ApplicationName as the thing the file name is "based on," it's still pointing at the wrong source.

I will fix this manually.

> The manifest is generated in the build output directory. By default, its file name is based on the app's <xref:Microsoft.Extensions.Hosting.IHostEnvironment.ApplicationName%2A> (for example, `{ASSEMBLY NAME}.staticwebassets.endpoints.json`, where the `{ASSEMBLY NAME}` placeholder is the app's assembly name). To provide a manifest from a different location, see the [Provide a custom static files manifest](#provide-a-custom-static-files-manifest) section.

Reviewer: ```suggestion
The manifest is generated in the build output directory at build time. Its file name is based on the project's assembly name (for example, `{ASSEMBLY NAME}.staticwebassets.endpoints.json`, where the `{ASSEMBLY NAME}` placeholder is the app's assembly name, the MSBuild `AssemblyName`). The assembly name usually matches the app's <xref:Microsoft.Extensions.Hosting.IHostEnvironment.ApplicationName%2A>, which defaults to the entry assembly name but is configurable. To provide a manifest from a different location, see the [Provide a custom static files manifest](#provide-a-custom-static-files-manifest) section.
```
Clarifying.

> The manifest is generated in the build output directory at build time. Its file name is based on the project's assembly name (for example, `{ASSEMBLY NAME}.staticwebassets.endpoints.json`, where the `{ASSEMBLY NAME}` placeholder is the app's assembly name, the MSBuild `AssemblyName`). The assembly name usually matches the app's <xref:Microsoft.Extensions.Hosting.IHostEnvironment.ApplicationName%2A>, which defaults to the entry assembly name but is configurable. To provide a manifest from a different location, see the [Provide a custom static files manifest](#provide-a-custom-static-files-manifest) section.

Reviewer: ```suggestion
The manifest is generated in the build output directory at build time. Its file name is based on the project's assembly name (for example, `{ASSEMBLY NAME}.staticwebassets.endpoints.json`, where the `{ASSEMBLY NAME}` placeholder is the app's MSBuild `AssemblyName` value). To provide a manifest from a different location, see the [Provide a custom static files manifest](#provide-a-custom-static-files-manifest) section.
```
Clarifying.
